### PR TITLE
(#1892) QA Runner

### LINKED ISSUE #1892: Ensure QA Test Suite is Executing on Jenkins Correctly
QA test suite needs to be configured and rolled out on continuous integration. At this point the tests take a long time and do not account for BlockBook servers. Requirements will need to be defined before moving forward with this.

### PR DESCRIPTION
**Pull request by hoffmabc**
Fixes #1892 

Now pulls in code that is triggering the QA test suite and also now exits out of the sub-scripts if there is a unit test failure.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,17 +25,51 @@ PKGMAP = $(P_TIMESTAMP),$(P_ANY)
...
 
+
+##
+## Testing
+##
+
+OPENBAZAARD_NAME ?= openbazaard-$(GIT_TAG)-$(GIT_SHA)
+BITCOIND_PATH ?= .
+
+.PHONY: openbazaard
```
Comment (tyler-smith) [inline Makefile]: This isn't a phony make target.

File: Dockerfile.dev [hunk 2]
```diff
@@ -1,28 +1,4 @@
...
-    pip3.6 install -r requirements.txt && \
-    wget https://bitcoin.org/bin/bitcoin-core-0.16.3/bitcoin-0.16.3-x86_64-linux-gnu.tar.gz && \
-    tar -xvzf bitcoin-0.16.3-x86_64-linux-gnu.tar.gz -C /opt
-
-RUN wget https://github.com/google/protobuf/releases/download/v3.6.0/protoc-3.6.0-linux-x86_64.zip && \
-    unzip ./protoc-3.6.0-linux-x86_64.zip -x readme.txt && \
-    mv ./include/* /usr/local/include/ && \
-    mv ./bin/protoc /usr/local/bin/ && \
-    rm -rf ./include ./bin
+FROM docker.dev.ob1.io/openbazaar-qa:0.10
```
Comment (hoffmabc) [inline Dockerfile.dev]: 👍 very nice
image

File: Makefile (hunk 1 above)
Comment (hoffmabc) [reply Makefile]: agreed

Comment (placer14) [reply Makefile]: I was concerned that the build output makes an `openbazaard-<GITTAG>-<GITSHA>` file... not `openbazaard`. Does that matter? (Not a makefile expert.)

File: Dockerfile.dev (hunk 2 above)
Comment (placer14) [reply Dockerfile.dev]: I wonder if anyone not authenticated against our registry will be able to pull this image.

Comment (hoffmabc) [reply Dockerfile.dev]: Not sure. @tyler-smith ?

**Comment by placer14** [comment]
Discussion with @szollo suggests that we change the docker repo that holds openbazaar-qa and openbazaar-dev to be on the public Docker Hub repo instead of our internal repo. Will apply that change and get another review on this.

**Comment by placer14** [comment]
Closing and opening to trigger jenkins build...

**Comment by placer14** [comment]
Trigger another build...

**Pull request merged**